src/commandLine/CommandLine.ts

```diff
@@ -0,0 +1,292 @@
+/* eslint-disable dot-notation */
+
+import * as pp from './printPretty';
+import * as help from './helpers';
+import { getArgs } from './args';
+const log = console.log;
+const RR = require('../modules/RokulRunnings').RokulRunnings;
+const Configstore = require('configstore');
+
+// get configs
+const rrConfig = new Configstore('Rokul Runnings', {
+  ip: '',
+  username: '',
+  password: '',
+  options: { pressDelay: 1000, retryDelay: 1000, retries: 1 },
+});
+
+// get args
+const args = getArgs();
+
+// set the parameters for starting the runner from the args passed in
+if (args.ip) {
+  rrConfig.set('ip', args.ip);
+}
+if (args.username) {
+  rrConfig.set('username', args.username);
+}
+if (args.password) {
+  rrConfig.set('password', args.password);
+}
+if (args.pressDelay || args.retryDelay || args.retries) {
+  const rrConfigOptions = rrConfig.get('options');
+  const rokuOptions = {
+    pressDelay: args.pressDelay ? args.pressDelay : rrConfigOptions.pressDelay,
+    retryDelay: args.retryDelay ? args.retryDelay : rrConfigOptions.retryDelay,
+    retries: args.retries ? args.retries : rrConfigOptions.retries,
+  };
+  rrConfig.set('options', rokuOptions);
+}
+
+// get parameters for starting the runner
+const ip = rrConfig.get('ip');
+const username = rrConfig.get('username');
+const password = rrConfig.get('password');
+const options = rrConfig.get('options');
+
+let runner;
+if (!(ip && username && password)) {
+  throw Error('IP, Username, or Password is missing. Please set before executing commands.');
+} else {
+  runner = new RR(ip, username, password, { ...options });
+}
+
+if (args.launchChannel) {
+  const opts = help.parseOpts({ opts: args.launchChannel, defaultOpt: 'channelCode' });
+  const okOpts = ['channelCode', 'contentId', 'mediaType', 'params'];
+  help.validateOpts(opts, okOpts);
+  Promise.resolve(log(runner.launchTheChannel(opts))).then(response => {
+    log(response);
+  });
+}
+
+if (args.deepLink) {
+  const opts = help.parseOpts({ opts: args.deepLink, defaultOpt: 'channelCode' });
+  const okOpts = ['channelCode', 'contentId', 'mediaType', 'params'];
+  help.validateOpts(opts, okOpts);
+  Promise.resolve(runner.deepLinkIntoChannel(opts)).then(response => {
+    log(response);
+  });
+}
+// Re-usable function for the basic get functions
+function getFunction(func) {
+  Promise.resolve(
+    func.then(response => {
+      pp.json(response);
+    }),
+  );
+}
+
+if (args.getApps) {
+  getFunction(runner.getApps());
+}
+
+if (args.getFocusedElement) {
+  getFunction(runner.getFocusedElement());
+}
+
+if (args.getScreenSource) {
+  getFunction(runner.getScreenSource());
+}
+
+if (args.getCurrentChannelInfo) {
+  getFunction(runner.getCurrentChannelInfo());
+}
+
+if (args.getDeviceInfo) {
+  getFunction(runner.getDeviceInfo());
+}
+
+if (args.getPlayerInfo) {
+  getFunction(runner.getPlayerInfo());
+}
+
+// This won't work if the folder where the screenshot should be saved doesn't exist
+if (args.getScreenshot) {
```

Comment:
each on these chunks looks really similar in structure. One thing you could consider is creating a function that hides the repetition.

Reply:
Fixed in 9262563